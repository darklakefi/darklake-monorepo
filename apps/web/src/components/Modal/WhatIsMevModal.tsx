import Modal, { ModalProps } from "@/components/Modal";

export default function WhatIsMevModal(props: ModalProps) {
  return (
    <Modal title="What is MEV?" {...props}>
      <div className="md:w-[430px]">
        <p className="uppercase text-body-2">
          MEV (MAXIMUM EXTRACTABLE VALUE) REFERS TO THE PROFITS THAT MINERS, VALIDATORS, OR ANYONE CONTROLLING
          TRANSACTION ORDERING CAN EXTRACT BY REORDERING, INCLUDING, OR EXCLUDING TRANSACTIONS WITHIN A BLOCK.
          <br />
          <br />
          ONE OF THE MOST COMMON FORMS OF MEV IS THE SANDWICH ATTACK. THIS HAPPENS WHEN A MALICIOUS ACTOR PLACES A TRADE
          RIGHT BEFORE AND AFTER A USER&#39;S TRADE TO MANIPULATE THE PRICE, EXTRACTING PROFIT IN THE PROCESS.
          <br />
          <br />
          In this tool, we calculate the MEV loss specifically from Sandwich Attacks on Raydium and Pump.fun.
        </p>
      </div>
    </Modal>
  );
}
