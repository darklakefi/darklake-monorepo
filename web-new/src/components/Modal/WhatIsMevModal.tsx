import Modal, { ModalProps } from "@/components/Modal";

export default function WhatIsMevModal(props: ModalProps) {
  return (
    <Modal title="What is MEV?" {...props}>
      <p className="uppercase">
        MEV (Maximum Extractable Value) refers to the profits that miners, validators, or anyone controlling transaction
        ordering can extract by reordering, including, or excluding transactions within a block.
        <br />
        <br />
        One of the most common forms of MEV is the <span className="text-brand-20">Sandwich Attack</span>. This happens
        when a malicious actor places a trade right before and after a user&#39;s trade to manipulate the price,
        extracting profit in the process.
        <br />
        <br />
        In this tool, we calculate the MEV loss specifically from Sandwich Attacks on Raydium and Pump.fun.
      </p>
    </Modal>
  );
}
