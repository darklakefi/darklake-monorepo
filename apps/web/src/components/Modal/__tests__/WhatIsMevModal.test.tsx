import { act, render, RenderResult } from "@testing-library/react";
import WhatIsMevModal from "@/components/Modal/WhatIsMevModal";

describe("<WhatIsMevModal />", () => {
  it("does not display content when closed", async () => {
    let baseElement: RenderResult["baseElement"] | undefined;

    await act(async () => {
      ({ baseElement } = render(<WhatIsMevModal isOpen={false} />));
    });

    expect(baseElement).toMatchSnapshot();
  });

  it("displays content successfully when open", async () => {
    let baseElement: RenderResult["baseElement"] | undefined;

    await act(async () => {
      ({ baseElement } = render(<WhatIsMevModal isOpen />));
    });

    expect(baseElement).toMatchSnapshot();
  });
});
