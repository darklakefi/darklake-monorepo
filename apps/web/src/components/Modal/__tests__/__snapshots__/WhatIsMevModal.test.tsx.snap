// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WhatIsMevModal /> displays content successfully when open 1`] = `
<body>
  <div />
  <div
    id="headlessui-portal-root"
  >
    <div
      data-headlessui-portal=""
    >
      <button
        aria-hidden="true"
        data-headlessui-focus-guard="true"
        style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
        type="button"
      />
      <div>
        <div
          aria-modal="true"
          class="relative"
          data-headlessui-state="open"
          data-open=""
          id="headlessui-dialog-:r0:"
          role="dialog"
          tabindex="-1"
        >
          <div
            aria-hidden="true"
            class="fixed inset-0 bg-modal-backdrop"
            data-headlessui-state="open"
            data-open=""
          />
          <div
            class="flex flex-col fixed inset-0 w-screen items-center justify-center"
          >
            <div
              data-headlessui-state="open"
              data-open=""
              id="headlessui-dialog-panel-:r7:"
            >
              <div
                class="py-4 pl-4 pr-7"
              >
                <div
                  class="max-h-[80vh] lg:max-h-none overflow-auto p-[16px] md:p-[40px] bg-brand-70 shadow-[12px_12px_0px_0px] shadow-brand-60 relative"
                >
                  <div
                    class="flex flex-row items-center mb-4 border-b border-b-brand-50 pb-3 justify-between"
                  >
                    <p
                      class="font-primary text-3xl leading-3xl text-brand-20"
                    >
                      What is MEV?
                    </p>
                    <button
                      class="hover-with-active flex items-center"
                      type="button"
                    >
                      <i
                        class="hn hn-times text-brand-30 text-xl leading-5"
                      />
                    </button>
                  </div>
                  <div
                    class="font-secondary text-lg leading-6 text-brand-30"
                  >
                    <div
                      class="md:w-[430px]"
                    >
                      <p
                        class="uppercase"
                      >
                        MEV (MAXIMUM EXTRACTABLE VALUE) REFERS TO THE PROFITS THAT MINERS, VALIDATORS, OR ANYONE CONTROLLING TRANSACTION ORDERING CAN EXTRACT BY REORDERING, INCLUDING, OR EXCLUDING TRANSACTIONS WITHIN A BLOCK.
                        <br />
                        <br />
                        ONE OF THE MOST COMMON FORMS OF MEV IS THE SANDWICH ATTACK. THIS HAPPENS WHEN A MALICIOUS ACTOR PLACES A TRADE RIGHT BEFORE AND AFTER A USER'S TRADE TO MANIPULATE THE PRICE, EXTRACTING PROFIT IN THE PROCESS.
                        <br />
                        <br />
                        <span
                          class="text-brand-20"
                        >
                          COMING SOON: THIS TOOL WILL CALCULATE MEV LOSS ANALYZING YOUR LAST 1000 TRANSACTIONS ON SOLANA.
                        </span>
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        aria-hidden="true"
        data-headlessui-focus-guard="true"
        style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px;"
        type="button"
      />
    </div>
  </div>
</body>
`;

exports[`<WhatIsMevModal /> does not display content when closed 1`] = `
<body>
  <div />
</body>
`;
